File: Assignment3/program5.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program5.c
//  Description : Accepts a character and checks it its vowel or not
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 10/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef int BOOL;
#define TRUE 1
#define FALSE 0

////////////////////////////////////////////////////////////////
// 
//  Function Name : CheckVowel
//  Description   : Checks the input character is a Vowel or not
//  Input         : Character
//  Output        : Boolean
//
////////////////////////////////////////////////////////////////

BOOL CheckVowel(char cVal)
{
    if(cVal == 'a' || cVal == 'e' || cVal == 'i' || cVal == 'o' || cVal == 'u' ||
        cVal == 'A' || cVal == 'E' || cVal == 'I' || cVal == 'O' || cVal == 'U')
        {
            return TRUE;
        }
        else
        {
            return FALSE;
        }
}

int main()
{
    char cValue = '\0';
    BOOL bRet = FALSE;

    printf("Enter Character\n");
    scanf("%c",&cValue);
    
    bRet = CheckVowel(cValue);

    if(bRet == 0)
    {
        printf("The Entered character %c is Not a Vowel(FALSE)\n",cValue);
    }
    else
    {
        printf("The Entered character %c is a Vowel(TRUE)\n",cValue);
    }
    
    return 0;
}

File: Assignment24/program5.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program5.java
//  Description : Accept a character character array coverting it into string and returns Reverse of that String
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : Reverse
//  Description   : Inputs character array coverting it into string and and returns Reverse of that String
//  Input         : String
//  Output        : String
//
////////////////////////////////////////////////////////////////

class Display
{
    public void Reverse(char[] chVal, int iSize)
    {
        int iCnt = 0;
        
        System.out.println("Reversed String : ");
        for(iCnt = (iSize - 1); iCnt >= 0; iCnt--)
        {
            System.out.print(chVal[iCnt]);
        }
    }
}

public class program5 
{
    public static void main(String args[])
    {
        Scanner sobj = new Scanner(System.in);

        String sValue = null;
        int iLength = 0;
        int iCnt = 0;
        char[] cValue = null;
        boolean bRet = false;

        System.out.println("Enter String : ");
        sValue = sobj.nextLine();

        iLength = sValue.length();

        cValue = new char[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            cValue[iCnt] = sValue.charAt(iCnt);
        }

        Display dobj = new Display();

        dobj.Reverse(cValue,iLength);

        sobj.close();
    }
}